[CI] testCorruptFileThenSnapshotAndRestore fails

Build failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/705/console

Reproduce command:

```
./gradlew :server:integTest \
  -Dtests.seed=7CC5E12B116EE7EF \
  -Dtests.class=org.elasticsearch.index.store.CorruptedFileIT \
  -Dtests.method="testCorruptFileThenSnapshotAndRestore" \
  -Dtests.security.manager=true \
  -Dtests.locale=es-NI \
  -Dtests.timezone=America/Puerto_Rico \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Log:

```
2-12T00:39:28,545][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] Index [221] docs async: [true] bulk: [false]
  1> [2018-12-12T00:39:28,547][WARN ][o.e.i.IndexSettings      ] [node_s0] [test] [index.merge.enabled] is set to false, this should only be used in tests and can cause serious problems in production environments
  1> [2018-12-12T00:39:28,549][INFO ][o.e.c.m.MetaDataMappingService] [node_s0] [test/eD6rpLjqTyC4pjUH0FwtvA] create_mapping [type]
  1> [2018-12-12T00:39:28,567][WARN ][o.e.i.IndexSettings      ] [node_s0] [test] [index.merge.enabled] is set to false, this should only be used in tests and can cause serious problems in production environments
  1> [2018-12-12T00:39:28,910][INFO ][o.e.t.CorruptionUtils    ] [testCorruptFileThenSnapshotAndRestore] Corrupting file --  flipping at position 907 from 72 to 73 file: _3.cfs
  1> [2018-12-12T00:39:28,912][INFO ][o.e.t.CorruptionUtils    ] [testCorruptFileThenSnapshotAndRestore] Checksum before: [999714073] after: [161080920] checksum value after corruption: 999714073] file: _3.cfs length: 2284
  1> [2018-12-12T00:39:28,912][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] --> shard [test][2], node[ONuHLlycR6OL81UlFk5aVA], [P], s[STARTED], a[id=XPSQ4UN1TDufxKK3h56rZA] has a corrupted file
  1> [2018-12-12T00:39:28,912][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] -->  creating repository
  1> [2018-12-12T00:39:28,928][INFO ][o.e.r.RepositoriesService] [node_s0] put repository [test-repo]
  1> [2018-12-12T00:39:28,963][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] --> snapshot
  1> [2018-12-12T00:39:28,990][INFO ][o.e.s.SnapshotsService   ] [node_s0] snapshot [test-repo:test-snap/hRZIxqy3RGm_t_khLuZA6w] started
  1> [2018-12-12T00:39:29,096][INFO ][o.e.s.SnapshotsService   ] [node_s0] snapshot [test-repo:test-snap/hRZIxqy3RGm_t_khLuZA6w] completed with state [SUCCESS]
  1> [2018-12-12T00:39:29,128][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] --> snapshot terminated with state SUCCESS
  1> [2018-12-12T00:39:29,138][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] [CorruptedFileIT#testCorruptFileThenSnapshotAndRestore]: cleaning up after test
  1> [2018-12-12T00:39:29,155][INFO ][o.e.c.m.MetaDataDeleteIndexService] [node_s0] [test/eD6rpLjqTyC4pjUH0FwtvA] deleting index
  1> [2018-12-12T00:39:29,238][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s0] removing template [random_index_template]
  1> [2018-12-12T00:39:29,258][INFO ][o.e.r.RepositoriesService] [node_s0] delete repository [test-repo]
  1> [2018-12-12T00:39:29,276][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] [CorruptedFileIT#testCorruptFileThenSnapshotAndRestore]: cleaned up after test
  1> [2018-12-12T00:39:29,276][INFO ][o.e.i.s.CorruptedFileIT  ] [testCorruptFileThenSnapshotAndRestore] after test
FAILURE 1.22s J7 | CorruptedFileIT.testCorruptFileThenSnapshotAndRestore <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <PARTIAL>
   >      but: was <SUCCESS>
   > 	at __randomizedtesting.SeedInfo.seed([7CC5E12B116EE7EF:B1115BAA8372AD8C]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.index.store.CorruptedFileIT.testCorruptFileThenSnapshotAndRestore(CorruptedFileIT.java:531)
   > 	at java.lang.Thread.run(Thread.java:748)
```

I believe this failure relates to soft-deletes (https://github.com/elastic/elasticsearch/pull/36141).

--- a/server/src/test/java/org/elasticsearch/index/store/CorruptedFileIT.java
+++ b/server/src/test/java/org/elasticsearch/index/store/CorruptedFileIT.java
@@ -475,7 +475,6 @@ public class CorruptedFileIT extends ESIntegTestCase {
      * TODO once checksum verification on snapshotting is implemented this test needs to be fixed or split into several
      * parts... We should also corrupt files on the actual snapshot and check that we don't restore the corrupted shard.
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/36526")
     public void testCorruptFileThenSnapshotAndRestore() throws ExecutionException, InterruptedException, IOException {
         int numDocs = scaledRandomIntBetween(100, 1000);
         internalCluster().ensureAtLeastNumDataNodes(2);
@@ -676,7 +675,10 @@ public class CorruptedFileIT extends ESIntegTestCase {
     private void pruneOldDeleteGenerations(Set<Path> files) {
         final TreeSet<Path> delFiles = new TreeSet<>();
         for (Path file : files) {
-            if (file.getFileName().toString().endsWith(".liv")) {
+            if (file.getFileName().toString().endsWith(".liv")
+            || file.getFileName().toString().endsWith(".dvm")
+            || file.getFileName().toString().endsWith(".dvd")
+            ) {
                 delFiles.add(file);
             }
         }
@@ -685,17 +687,9 @@ public class CorruptedFileIT extends ESIntegTestCase {
             if (last != null) {
                 final String newSegmentName = IndexFileNames.parseSegmentName(current.getFileName().toString());
                 final String oldSegmentName = IndexFileNames.parseSegmentName(last.getFileName().toString());
-                if (newSegmentName.equals(oldSegmentName)) {
-                    int oldGen =
-                        Integer.parseInt(
-                            IndexFileNames.stripExtension(
-                                IndexFileNames.stripSegmentName(last.getFileName().toString())).replace("_", ""),
-                            Character.MAX_RADIX
-                        );
-                    int newGen = Integer.parseInt(
-                        IndexFileNames.stripExtension(
-                            IndexFileNames.stripSegmentName(current.getFileName().toString())).replace("_", ""),
-                        Character.MAX_RADIX);
+                if (newSegmentName.equals(oldSegmentName) ) {
+                    long oldGen = IndexFileNames.parseGeneration(last.getFileName().toString());
+                    long newGen = IndexFileNames.parseGeneration(current.getFileName().toString());
                     if (newGen > oldGen) {
                         files.remove(last);
                     } else {

--- a/test/framework/src/main/java/org/elasticsearch/test/CorruptionUtils.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/CorruptionUtils.java
@@ -93,10 +93,12 @@ public final class CorruptionUtils {
 
             long checksumAfterCorruption;
             long actualChecksumAfterCorruption;
+
             try (ChecksumIndexInput input = dir.openChecksumInput(fileToCorrupt.getFileName().toString(), IOContext.DEFAULT)) {
                 assertThat(input.getFilePointer(), is(0L));
-                input.seek(input.length() - 8); // one long is the checksum... 8 bytes
+                input.seek(input.length() - CodecUtil.footerLength());
                 checksumAfterCorruption = input.getChecksum();
+                input.seek(input.length() - 8);
                 actualChecksumAfterCorruption = input.readLong();
             }
             // we need to add assumptions here that the checksums actually really don't match there is a small chance to get collisions

